### PR TITLE
fix(core/highlight): wrap pre content's in code, per HTML spec

### LINKED ISSUE #2084: Modules should contain their own localization information
We currently have localization strings in core/l10n.js that are used throughout ReSpec. However, it's rare that a localized string is used in more than one place at a time. What would be better is for where localization happens, the strings be included in the appropriate module itself. 

Then core/l10n should only export the `lang` of the document, which should be configurable. Then each module that requires localization should just key off that value.  

This model is already used by the linter modules:
https://github.com/w3c/respec/blob/develop/src/core/linter-rules/check-charset.js#L6-L18

So, we just need to copy that. 

The following remain:

- [x] src/core/figures.js
- [x] src/core/github.js
- [x] src/core/issues-notes.js
- [x] src/core/structure.js
- [x] src/w3c/headers.js
- [x] src/w3c/templates/cgbg-headers.js
- [x] src/w3c/templates/cgbg-sotd.js
- [x] src/w3c/templates/headers.js
- [x] src/w3c/templates/sotd.js

### PR DESCRIPTION
**Pull request by marcoscaceres**
fixes #2084 
close #2111

### CODE REVIEW

File: tests/spec/core/highlight-spec.js [hunk 1]
```diff
@@ -74,6 +105,79 @@ describe("Core — Highlight", () => {
...
+              Header: Test5
+            </code>
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const firstPre = doc.getElementById("first-pre");
+    expect(firstPre.textContent).toContain(
+      "this function(){} is not highlighted"
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]: Wouldn't this be always true even after highlighting? Should we check `innerHTML` instead?

File: tests/spec/core/highlight-spec.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ describe("Core — Highlight", () => {
     const doc = await makeRSDoc(ops);
     const pre = doc.querySelector("pre");
     expect(pre.classList.contains("hljs")).toBeFalsy();
-    expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
+    expect(pre.querySelectorAll("span[class~=hljs-]").length).toBe(0);
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/highlight-spec.js]: agree

File: tests/spec/core/highlight-spec.js [hunk 3]
```diff
@@ -53,18 +53,49 @@ describe("Core — Highlight", () => {
               alert('foo');
             }
           </pre>
+          <pre>
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/highlight-spec.js]: nope. Oops.

File: tests/spec/core/highlight-spec.js (hunk 1 above)
Comment (marcoscaceres) [reply tests/spec/core/highlight-spec.js]: yes, you are right.

File: tests/spec/core/highlight-spec.js [hunk 4]
```diff
@@ -53,18 +53,49 @@ describe("Core — Highlight", () => {
...
+              function(){}
+            </code>
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const codeNoHighlight = doc.querySelector("div.example code.nohighlight");
+    expect(codeNoHighlight).toBeTruthy();
+    expect(codeNoHighlight.querySelectorAll("span[class*=hljs-]").length).toBe(
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply tests/spec/core/highlight-spec.js]: Did the same for the others... this is much nicer than .greaterThan()

File: tests/spec/core/highlight-spec.js [hunk 5]
```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
...
+              alert('foo');
+            }
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+    const pre = doc.querySelector("pre");
+    const code = pre.firstElementChild;
+    expect(code.localName).toBe("code");
+    expect(code.classList.contains("hljs")).toBeTruthy();
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(code.classList).toContain("hljs");
```

File: tests/spec/core/highlight-spec.js [hunk 6]
```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
...
+    expect(code.classList.contains("js")).toBeTruthy();
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(code.classList).toContain("js");
```

File: tests/spec/core/highlight-spec.js [hunk 7]
```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
...
+              Header: Test5
+            </code>
+          </pre>
+        </section>`,
+    };
+    const doc = await makeRSDoc(ops);
+
+    const firstPre = doc.getElementById("first-pre");
+    expect(firstPre.innerHTML).toContain("function pass(){}");
+    expect(firstPre.querySelectorAll("code span[class*=hljs-]")).toBeTruthy();
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(firstPre.querySelector("code span[class*=hljs-]")).toBeTruthy();
```

File: tests/spec/core/highlight-spec.js [hunk 8]
```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
...
+    const secondPre = doc.getElementById("second-pre");
+    expect(secondPre.innerHTML).toContain(
+      "second function(){} is not highlighted"
+    );
+
+    const fourthCode = secondPre.querySelector("code:nth-child(2)");
+    expect(fourthCode.classList.contains("javascript")).toBeTruthy();
+    const lastCode = secondPre.querySelector("code:last-child");
+    expect(lastCode.textContent).toContain("Header: Test5");
+    expect(lastCode.classList.contains("http")).toBeTruthy();
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(lastCode.classList).toContain("http");
```

File: tests/spec/core/highlight-spec.js [hunk 9]
```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
     };
     const doc = await makeRSDoc(ops);
     const pre = doc.getElementById("test");
-    expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
+    expect(pre.querySelectorAll("span[class*=hljs-]").length).toBe(0);
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(pre.querySelector("span[class*=hljs-]")).toBeNull();
```

File: tests/spec/core/highlight-spec.js [hunk 10]
```diff
@@ -58,13 +56,39 @@ describe("Core — Highlight", () => {
     const doc = await makeRSDoc(ops);
     const pre = doc.querySelector("div.example pre");
     expect(pre.classList.contains("nohighlight")).toBeTruthy();
-    expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
+    expect(pre.querySelectorAll("span[class*=hljs-]").length).toBe(0);
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(pre.querySelector("span[class*=hljs-]")).toBeNull();
```

File: tests/spec/core/highlight-spec.js (hunk 8 above)
Comment (marcoscaceres) [reply tests/spec/core/highlight-spec.js]: Didn't think it would be smart enough to do this.

**Comment by marcoscaceres** [comment]
@saschanaz, I think that's all of them. Thanks for catching all those little things.

File: tests/spec/core/highlight-spec.js [hunk 11]
```diff
@@ -74,6 +98,76 @@ describe("Core — Highlight", () => {
...
+      "function one(){}"
+    );
+
+    const secondPre = doc.getElementById("second-pre");
+    expect(secondPre.innerHTML).toContain(
+      "second function(){} is not highlighted"
+    );
+
+    const fourthCode = secondPre.querySelector("code:nth-child(2)");
+    expect(fourthCode.classList.contains("javascript")).toBeTruthy();
```
Comment (saschanaz) [inline tests/spec/core/highlight-spec.js]:
```suggestion
    expect(fourthCode.classList).toContain("javascript");
```

**Comment by marcoscaceres** [comment]
> I think it would also be good to test removing language hint class from <pre>, which seems to be lacking currently.

**Comment by marcoscaceres** [comment]
> I think it would also be good to test removing language hint class from <pre>, which seems to be lacking currently.

on it

**Comment by marcoscaceres** [comment]
Just want to acknowledge @CodHeK. He did most of the groundwork here and lead us to ask some sensible questions. That lead to a good refactor, which hopefully will make this module more maintainable.

**Pull request merged**